mail/isync/Portfile

```diff
@@ -20,23 +20,22 @@ long_description    isync is a command line utility for synchronizing a \
 homepage            http://isync.sourceforge.net/
 master_sites        sourceforge:project/isync/isync/${version}/
 
-checksums           rmd160  65ce1693e2eec5619fc9648e6c568eb4bf9679be \
-                    sha256  68cb4643d58152097f01c9b3abead7d7d4c9563183d72f3c2a31d22bc168f0ea \
-                    size    311868
+checksums           rmd160  6f37d53e9e779dd2ccbd1a862e43fb9065e064e4 \
+                    sha256  9106d1428c912f421a659a22c3c4dbe9fe110f3f4da1498038e6ebf8d284e805 \
+                    size    314856
 
 depends_build       port:perl5 \
                     port:pkgconfig
 
-depends_lib         port:db53 \
+depends_lib         port:cyrus-sasl2 \
+                    port:db53 \
                     path:lib/libssl.dylib:openssl \
                     port:zlib
 
 depends_run         path:share/curl/curl-ca-bundle.crt:curl-ca-bundle
 
-patchfiles          patch-src-drv_proxy_gen.pl.diff
-
 post-patch {
-    reinplace "s|@PREFIX@|${prefix}|g" ${worksrcpath}/src/drv_proxy_gen.pl
+    reinplace "s|#!/usr/bin/perl|#!${prefix}/bin/perl|" ${worksrcpath}/src/drv_proxy_gen.pl
```

Comment:
Thank you for the update. Could we keep the patchfile and this reinplace the way it was, please?

Reply:
Could you please explain why?

Reply:
If, when updating the port to a new version, the original file has changed to such a degree that the patch fails to apply, that will stop the installation, requiring the developer to inspect the file and figure out how the patch needs to be adjusted or possibly removed. That's good: it requires the developer to address the issue before they're able to install the revised port. The context in a patchfile makes it clear what's being replaced and helps the developer know how to make a new similar patch.

On the other hand, just using a reinplace provides no context, and a failing reinplace only generates a warning; it does not stop the build. (Maybe we should change that, but for now, it's just a warning.) It would be possible for someone to update the port, for the reinplace to fail to apply, for the maintainer to ignore or not see the warning, see that the port installed, and commit the update; meanwhile the change that the reinplace was supposed to make is no longer being made, which will probably mean the port won't work as intended.

Reply:
Thank you, I was not aware that reinplace does not stop the build on failure. IMO, the reinplace behaviour should be changed; It is rather cumbersome that one needs to introduce a patch for every reinplace usage because of this. I agree that in some cases a patch may provide more context; however, in this case it should be evident from the reinplace statement that it is supposed to modify a shebang. BTW, the isync Portfile includes another reinplace with no corresponding patch. Is there any other way in a Portfile to stop on reinplace failure that does not require adding a patch?

Reply:
> Thank you, I was not aware that reinplace does not stop the build on failure. IMO, the reinplace behaviour should be changed;

I've filed https://trac.macports.org/ticket/60844 for that.

> It is rather cumbersome that one needs to introduce a patch for every reinplace usage because of this.

It's not intended to be cumbersome; it's intended to be helpful for future maintainers. The developer who writes a complicated regex in a reinplace may understand exactly what it does and why it does it, but once that developer leaves MacPorts and it falls to someone else to update the port, it can be quite difficult to understand what was intended. A patchfile makes the intention clearer. A reinplace by itself should only be used when the equivalent patchfile would be prohibitively large (e.g. replacing the same thing in thousands of files), or when the value of a variable needs to be used.

> I agree that in some cases a patch may provide more context; however, in this case it should be evident from the reinplace statement that it is supposed to modify a shebang.

You're probably right that in this case the context isn't needed to understand what's going on. However since we do still want to ensure that the replacement takes place, I recommend sticking with the patchfile.

> BTW, the isync Portfile includes another reinplace with no corresponding patch.

Yes, I see that. My preference would be for that to use a patchfile too. Note that since both of the patches would require replacement of the same variable value, that simplifies things, because you only need a single reinplace statement to modify any number of files.

> Is there any other way in a Portfile to stop on reinplace failure that does not require adding a patch?

You could manually write the code to do that. For example, copy the file before the reinplace, then do the reinplace, then `cmp` the old file and the new file and error out if they are the same. But that would be unusual; I don't think I've seen any other ports do that.

Reply:
> I've filed https://trac.macports.org/ticket/60844 for that.

Great, thank you.

> Yes, I see that. My preference would be for that to use a patchfile too. Note that since both of the patches would require replacement of the same variable value, that simplifies things, because you only need a single reinplace statement to modify any number of files.

Good point. I changed the replacement logic as you suggested, does it look good?

Reply:
ping?